### PR TITLE
[docs] fixtures and Python tests

### PR DESCRIPTION
**Pull request by alfredodeza**
describe layout of Python tests and fixtures, with links to external resources for `testinfra` and `pytest`

### CODE REVIEW

File: docs/source/testing/layout.rst [hunk 1]
```diff
@@ -32,6 +32,22 @@ At the very least, a scenario will need these files:
...
 -----------
+Python test files (unlike scenarios) rely on paths to *map* where they belong. For
+example, a file that should only test monitor nodes would live in
+``ceph-ansible/tests/functional/tests/mon/``. Internally, the test runner
+(``py.test``) will *mark* these as tests that should run on a monitor only.
+Since the configuration of a scenario already defines what node has a given
+role, then it is easier for the system to only run tests that belong to
+a particular node type.
+
+The current convention is a bit manual, with initiall path support for:
```
Comment (andrewschoen) [inline docs/source/testing/layout.rst]: s/initiall/initial

File: docs/source/testing/tests.rst [hunk 2]
```diff
@@ -2,9 +2,93 @@
...
+
+For remote execution, we can rely further on other fixtures (tests can have as
+many fixtures as needed) like ``File``::
+
+    def test_ceph_config_has_inital_members_line(self, node, File):
+        assert File(node["conf_path"]).contains("^mon initial members = .*$")
+
+
+For the full list of available fixtures
+.. note:: For a full list of other fixtures avai
```
Comment (andrewschoen) [inline docs/source/testing/tests.rst]: Did you mean to link to testinfra here?

Comment (alfredodeza) [inline docs/source/testing/tests.rst]: geez, yes I meant that

File: docs/source/testing/layout.rst (hunk 1 above)
Comment (alfredodeza) [inline docs/source/testing/layout.rst]: I was trying to include all the inits, so initiall felt natural here

**Comment by andrewschoen** [comment]
jenkins test docs

**Pull request merged**